### PR TITLE
feat(spanner): allow custom lib name and version for telemetry purpose

### PR DESCRIPTION
**Pull request by jiren**
Towards: #4761 

Allow user-agent to be passed in and not a hardcoded version. This will helpful for database connector i.e ActiveRecord connector.

### CODE REVIEW

File: google-cloud-spanner/lib/google-cloud-spanner.rb [hunk 1]
```diff
@@ -45,6 +45,12 @@ module Cloud
     # @param [Integer] timeout Default timeout to use in requests. Optional.
     # @param [Hash] client_config A hash of values to override the default
     #   behavior of the API client. Optional.
+    # @param [String] lib_name Library name. This will override the default
+    #   name of the library in the api call request header
+    #   for telemetry. Optional.
```
Comment (skuruppu) [inline google-cloud-spanner/lib/google-cloud-spanner.rb]: Reformat please.

File: google-cloud-spanner/lib/google/cloud/spanner.rb [hunk 2]
```diff
@@ -68,6 +68,12 @@ module Spanner
       #   Deprecated.
       # @param [String] emulator_host Spanner emulator host. Optional.
       #   If the param is nil, uses the value of the `emulator_host` config.
+      # @param [String] lib_name Library name. This will override the default
+      #   name of the library in the api call request header
+      #   for telemetry. Optional.
```
Comment (skuruppu) [inline google-cloud-spanner/lib/google/cloud/spanner.rb]: This needs reformatting as well.

**Comment by jiren** [comment]
> This seems reasonable. But I wonder if the user agent should also contain which version of the client lib is being used by the caller. E.g. should it be:
> 
> x-goog-api-client':
> 'spanner-activerecord/0.0.1 google-cloud-spanner/1.13.1 gl-ruby/ grpc/<gRPC_VERSION> gax/<GAX_VERSION>'
> 
> Also, minor nit, please reformat the doc strings added in this PR as the line-wrapping seems a bit off.

@skuruppu  Updated code and added a prefix to lib name while creating underline gapic client instance.

**Comment by odeke-em** [comment]
Thank you @skuruppu for the review! @jiren please rebase from master as there is a merge conflict, but also @skuruppu gave you a LGTM.

**Pull request merged**